os/mm/kmm_heap/kmm_free.c

```diff
@@ -92,8 +92,12 @@
 
 void kmm_free(FAR void *mem)
 {
-	struct mm_heap_s *kheap = mm_get_heap(mem);
-	DEBUGASSERT(kmm_heapmember(mem));
-	mm_free(kheap, mem);
+	struct mm_heap_s *kheap;
+
+	if (mem) {
+		kheap = mm_get_heap(mem);
+		DEBUGASSERT(kmm_heapmember(mem));
```

Comment:
How about changing the position between line 98 and 99?
We don't need to call mm_get_heap if mem is not in kernel heap.